Android/Self_Assessment_Devarshi/Intermediate/app/src/main/java/com/example/intermediate/SecondFragment.java

```diff
@@ -0,0 +1,175 @@
+        dropdown.setAdapter(adapter);
+        Spinner s=view.findViewById(R.id.spinner1);
+        s.setSelection(adapter.getPosition(t.getType()));
+
+        Departure.setInputType(InputType.TYPE_NULL);
+        final Calendar[] date = new Calendar[1];
+        Departure.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.N)
+            @Override
+            public void onClick(View v) {
+                final Calendar currentDate = Calendar.getInstance();
+                date[0] =  Calendar.getInstance();
+                new DatePickerDialog(SecondFragment.super.getContext(), new DatePickerDialog.OnDateSetListener() {
+                    @Override
+                    public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
+                        date[0].set(year, monthOfYear, dayOfMonth);
+                        new TimePickerDialog(SecondFragment.super.getContext(), new TimePickerDialog.OnTimeSetListener() {
+                            @Override
+                            public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
+                                date[0].set(Calendar.HOUR_OF_DAY, hourOfDay);
+                                date[0].set(Calendar.MINUTE, minute);
+                                android.text.format.DateFormat df = new android.text.format.DateFormat();
+                                SimpleDateFormat dateFormat = new SimpleDateFormat(  "yyyy-MM-dd HH:mm:ss", Locale.getDefault());
+                                Departure.setText( dateFormat.format(date[0]));
+
+                            }
+                        }, currentDate.get(Calendar.HOUR_OF_DAY), currentDate.get(Calendar.MINUTE), false).show();
+                    }
+                }, currentDate.get(Calendar.YEAR), currentDate.get(Calendar.MONTH), currentDate.get(Calendar.DATE)).show();
+            }
+
+        });
+
+        delete=view.findViewById(R.id.delete);
+        update=view.findViewById(R.id.update);
+
+        delete.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                performDelete(v);
+            }
+        });
+        update.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                performUpdate(v);
+            }
+        });
+
+        return view;
+    }
+    public void performDelete(View view){
+        db.collection("todos").document(id)
+                .delete()
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        Log.d("TAG", "DocumentSnapshot successfully deleted!");
+                        FirstFragment someFragment = new FirstFragment();
+                        FragmentTransaction transaction = getFragmentManager().beginTransaction();
```

Comment:
getFragmentManager() is deprecated, should use other recommended API. Also, try using lambdas instead of anonymous listeners all over the usage in code.

Reply:
Done